### PR TITLE
[DFMS-220] Generate variant and env labels dynamically

### PR DESCRIPTION
**Pull request by rexagod**
Generate variant and env labels dynamically, in addition to cluster ids
as well, since unlike telemeter, these are not added by default in
observatorium, refer: https://coreos.slack.com/archives/C010A637DGB/p1658834656153389?thread_ts=1658822806.667269&cid=C010A637DGB

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>
***
**This PR is rebased on top of #65, so that needs to go in first.**
***
Changes: [`f280c9b` (#69)](https://github.com/red-hat-storage/mcg-osd-deployer/pull/69/commits/f280c9b4cf1a92439e74b21675922e2a13ecdf3c)

### CODE REVIEW

File: controllers/monitoring_alert.go [hunk 1]
```diff
@@ -167,6 +176,38 @@ func (r *ManagedMCGReconciler) reconcileNoobaaRules() error {
 			return fmt.Errorf("failed to unmarshal noobaa-rules.yaml: %w", err)
 		}
 
+		// addLabel is a helper function to add a particular label to all rules in the PrometheusRuleSpec.
+		addLabel := func(label, value string) {
```
Comment (jmolmo) [inline controllers/monitoring_alert.go:180]: I think that using the `external_labels` setting in prometheus configuration would fit very well with our requirements, (and less code is always better.) In any case your proposal is ok  for me.

https://prometheus.io/docs/prometheus/latest/configuration/configuration/ 
```
# The labels to add to any time series or alerts when communicating with
# external systems (federation, remote storage, Alertmanager).
external_labels:
    [ [<labelname>]: [<labelvalue>]
```

**Review by jmolmo — APPROVED** [review]
LGTM

File: controllers/monitoring_alert.go (hunk 1 above)
Comment (rexagod) [reply controllers/monitoring_alert.go:180]: Tracking this here: https://github.com/red-hat-storage/mcg-osd-deployer/issues/72. Will come back to it and replace this approach if the `external_labels` thing fulfills our use case once I have a bit more b/w in the future.

**Review by naveenpaul1 — APPROVED** [review]
lgtm

**Pull request merged**